### PR TITLE
DOC: Document that datetime64 is a dtype. 

### LINKED ISSUE #10039: datetime64 not present in intersphinx inventory
Hello, over at Matplotlib we're trying to reference your docs on `datetime64` objects, but it doesn't look like there is an entry in the numpy intersphinx inventory. Running the following doesn't produce any output:

```bash
python -m sphinx.ext.intersphinx 'https://docs.scipy.org/doc/numpy/objects.inv' | grep 'datetime64'
```

The grep can be removed to see the whole numpy intersphinx inventory. xref https://github.com/matplotlib/matplotlib/issues/9802

### PR DESCRIPTION
**Pull request by dstansby**
Fixes #10039

### CODE REVIEW

File: numpy/core/_add_newdocs.py [hunk 1]
```diff
@@ -6831,6 +6831,11 @@ def add_newdoc_for_scalar_type(obj, fixed_aliases, doc):
     numbers.
     """)
 
+add_newdoc_for_scalar_type('datetime64', [''],
+    """
+    Datetime type.
```
Comment (charris) [inline numpy/core/_add_newdocs.py]: This could use a longer explanation. Look at some of the other dtype docstrings in this file.

Comment (dstansby) [reply numpy/core/_add_newdocs.py]: Happy if someone wants to suggest a better description, but I'm struggling to come up with anything myself.

**Comment by eric-wieser** [comment]
I don't think this is sufficient to create an intersphinx link - no doc page is emitted for any of the scalar types.

File: numpy/core/_add_newdocs.py (hunk 1 above)
Comment (mattip) [reply numpy/core/_add_newdocs.py]: "Datetime type, stored as a 64-bit integer. Functionally compatible with `datetime.datetime`"

I think single backticks around `datetime.datetime` will create a link to the python documentation. Should also document timedelta64

Comment (mattip) [reply numpy/core/_add_newdocs.py]: @dstansby ping

Comment (dstansby) [reply numpy/core/_add_newdocs.py]: Afraid I don't have the bandwidth to pick this up at the moment - if someone else wants to push to my branch or open a replacement PR then feel free.

**Pull request closed without merging**